src/main/java/com/blackducksoftware/integration/alert/common/descriptor/ProviderDescriptor.java

```diff
@@ -39,4 +41,6 @@ public ProviderDescriptor(final String name, final DescriptorConfig providerDesc
     public Provider getProvider() {
         return provider;
     }
+
+    public abstract Set<String> getNotificationTypes();
```

Comment:
should this call provider.getNotificationTypes instead of being abstract? Or will this have a different purpose?